src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1536,7 +1536,7 @@ get_shard_id_for_distribution_column(PG_FUNCTION_ARGS)
 						errmsg("relation is not distributed")));
 	}
 
-	if (IsCitusTableType(relationId, CITUS_TABLE_WITH_NO_DIST_KEY))
+	if (IsCitusTableType(relationId, CITUS_LOCAL_OR_REFERENCE_TABLE))
```

Comment:
I don't quite understand how this change helps us with single shard tables.

Wouldn't a lot of the cases where we use CITUS_TABLE_WITH_NO_DIST_KEY apply to single shard tables in the same way?

Reply:
Now that we want to interpret null dist key tables as distributed, IsCitusTableType(DISTRIBUTED_TABLE) should return true for such tables too.

However, it would be confusing if IsCitusTableType(CITUS_TABLE_WITH_NO_DIST_KEY) also returns true for the same table. Because in that case, even the order of `if` conditions would be prone to bugs:
```c
if (IsCitusTableType(DISTRIBUTED_TABLE))
{
  // then it's distributed table
}
else if (IsCitusTableType(CITUS_TABLE_WITH_NO_DIST_KEY))
{
..
}

// vs

if (IsCitusTableType(CITUS_TABLE_WITH_NO_DIST_KEY))
{
  // then it's not a distributed table
}
else if (IsCitusTableType(DISTRIBUTED_TABLE))
{
..
}
```

For this reason, I replaced CITUS_TABLE_WITH_NO_DIST_KEY with CITUS_LOCAL_OR_REFERENCE_TABLE and scoped its definition down to citus local tables / reference tables. For the codepaths where it makes more sense to act similar to citus local tables / reference tables, I'm planning to add necessary `or` conditions in future PRs.

Reply:
That depends on whether you interpret CITUS_TABLE_WITH_NO_DIST_KEY as a table type or a table property.

I mostly interpret it as a table property, and the logic here makes sense to me for tables with that property.